src/api/lib/tasks/dev.rake

```diff
@@ -99,6 +99,62 @@ namespace :dev do
       end
     end
   end
+  namespace :testdata do
+    task create: :environment do
+      unless Rails.env.to_s == 'development'
+        puts "You are running this rake task in #{Rails.env} environment."
+        puts 'Please only run this task with RAILS_ENV=development'
+        puts 'otherwise it will destroy your database data.'
+        return
+      end 
+      require 'factory_bot'
+      include FactoryBot::Syntax::Methods
+      
+      Rake::Task['db:drop'].invoke
+      Rake::Task['db:create'].invoke
+      Rake::Task['db:setup'].invoke
+      Rake::Task['db:seed'].invoke
+
+      iggy = create(:confirmed_user, login: 'Iggy')
+      admin = User.where(login: 'Admin').first
+      User.current = admin
+
+      interconnect = create(:project, name: 'openSUSE.org', remoteurl: 'https://api.opensuse.org/public')
+      tw_repository = create(:repository, name: 'snapshot', project: interconnect, remote_project_name: 'openSUSE:Factory')
+
+      home_admin = create(:project, name: admin.home_project_name)
+      admin_repository = create(:repository, project: home_admin, name: 'openSUSE_Tumbleweed')
+      create(:path_element, link: tw_repository, repository: admin_repository)
+      branches_admin = create(:project, name: admin.branch_project_name('home:Iggy'))
+      ruby_admin = create(:package_with_file, name: 'ruby', project: branches_admin, file_content: 'from home')
+
+      home_iggy = create(:project, name: iggy.home_project_name)
+      ruby_iggy = create(:package_with_file, name: 'ruby', project: home_iggy, file_content: 'from iggies branch')
+      create(
+        :bs_request_with_submit_action,
+        creator: admin,
+        target_project: home_iggy,
+        target_package: ruby_iggy,
+        source_project: branches_admin,
+        source_package: ruby_admin
+      )
+
+      leap = create(:project, name: 'openSUSE:Leap:15.0')
+      create(:package_with_file, name: 'apache2', project: leap)
+      leap_repository = create(:repository, project: leap, name: 'openSUSE_Tumbleweed')
+      create(:path_element, link: tw_repository, repository: leap_repository)
+
+      User.current = admin
```

Comment:
Is there a reason for doing this again?

Reply:
Yeah, some of the factories set the User.current back to nil :cry:

Reply:
Oh... alright. I would add a comment to explain this. Changing the factories is also another solution, but it implies a lot of changes if many specs depend on this behavior.